tests/scalar_advection/scalar_advection.py

```diff
@@ -1,6 +1,6 @@
 # Demo for pure scalar advection - this is adapted from the Firedrake DG_advection demo,
-# but here we use G-ADOPT's Energy solver and use a CG discretisation
-# with Streamline Upwind (SU) stabilisation.
+# but here we use G-ADOPT's GenericTransportSolver and use a CG discretisation with
```

Comment:
This maybe semantics on my part, but we are still solving the energy equation here, right, albeit with SU stabilisation? Why then are we switching to Generic Transport solver? Or is that just to ensure that it is tested? If the latter, could we also ensure that the energy solver is tested (and configurable) with SU?